CRM/Core/Payment/PayPalProIPN.php

```diff
@@ -419,7 +409,7 @@ public function single($input, $contribution, $recur = FALSE, $first = FALSE) {
       return;
     }
 
-    CRM_Contribute_BAO_Contribution::completeOrder($input, $this->getContributionRecurID(), $contribution->id ?? NULL);
+    CRM_Contribute_BAO_Contribution::completeOrder($input, $this->getContributionRecurID(), $this->getContributionID());
```

Comment:
Fine

Reply:
See how repeat other never gets the contribution - so that invoice_id is set but discarded